### ISSUE TITLE
Cypress fails when visiting a site with Chatbase

### ISSUE THREAD

**Issue by YakovL** [cypress-io/cypress#27437]
### Current behavior

At some point, e2e testing via Cypress of our site started to fail on the first step, visiting the site itself. The error that Cypress reports in the output is:

> ReferenceError: The following error originated from your test code, not from Cypress.
> 
>   > process is not defined
> 
> When Cypress detects uncaught errors originating from your test code it will automatically fail the current test.

I did some more experiments and found that it also fails when visiting `cy.visit('https://www.chatbase.co');` which we've integrated at approximately the time of Chatbase integration.

The errors displayed in the Cypress UI are `CypressError: cy.task() must only be invoked from the spec file or support file`:

![image](https://github.com/cypress-io/cypress/assets/1131924/dd9a5017-3acb-4174-853e-f1b41b8ab971)

Which seems to be somewhat irrelevant (or at least unhelpful) as the same test works for pages without Chatbase (as another suspect, I've tested with `cy.visit('https://marker.io');` which doesn't produce the error).

### Desired behavior

Well, I'm expecting no problems like that (or at very least better reporting of the problem).

### Test code to reproduce

Note: I'm also using `@badeball/cypress-cucumber-preprocessor`, `@cucumber/cucumber`, and `@synthetixio/synpress`, but I haven't reproduced the bug without them yet.

1. Clone the https://github.com/YakovL/synpress-cucumber repo;
2. `git checkout chatbase-problem`;
3. `npm i` (probably `yarn` will also suffice);
4. `npm run test:e2e`.

You'll see the errors in the report. Note that `@synthetixio/synpress` breaks the Cypress watch mode, so you won't be able to do `cypress open` (well, you can, but you'll get [other errors](https://github.com/Synthetixio/synpress/issues/775) irrelevant for this case).

### Cypress Version

12.17.2

### Node version

18.16.0

### Operating System

Windows 10 Pro x64 (22H2, 19045.3086)

### Debug Logs

I've failed to make a copy of the log as both `> filename` and `Start-Transcript` haven't redirected it file. I hope it will be reproduced using the repo linked above

### Other

I'll do my best to provide additional details in the issue thread, if needed

**Comment by nagash77** [comment]
@YakovL it would be helpful to know if you have changed versions of cypress and what changes have been made to the AUT to narrow down your problem.  Debug logs would also be helpful.

It's a shot in the dark but you may want to try turning on [modify obstructive third party code flag](https://docs.cypress.io/guides/guides/web-security#Modifying-Obstructive-Third-Party-Code) to see if that helps.

Also check out the Cypress Discord Community [![Discord chat](https://img.shields.io/badge/chat-on%20Discord-brightgreen)](https://on.cypress.io/discord), it can be helpful for debugging or answering questions on how to use Cypress.

**Comment by YakovL** [comment]
Ok, I've made one step further to narrowing down the problem: switched from `.feature` files to `.cy.ts` files (updated the branch in the linked repo). I'll try to narrow it even further, but preliminary results are:
* removing `addCucumberPreprocessorPlugin` from config doesn't remove the issue
* removing either `synpressPlugins` (with `import '@synthetixio/synpress/support/index';` in e2e.ts) or `file:preprocessor` (involving `createBundler` and `createEsbuildPlugin`) removes the issue

**Comment by YakovL** [comment]
@nagash77 answering some of your questions:

As for changing the Cypress versions, initially I've started from 12.13.0 (as far as I remember), than because of some bugs I've updated it to 12.17.0, than to 12.17.2.

To narrow down the problem, I've compared tests of our app (failing on `cy.visit`) with my "reference" autotests in the linked repo and then (the problem wasn't there, but was reproduced once I switched visiting another app to visiting our app) I've suspected that the issue is caused by one of the integrations we used. So I've checked the pages of the integrations themselves, and the bug was reproduced on the www.chatbase.co. There was no changes in UAT that caused the problem.
